packages/relationships/src/utils/query.ts

```diff
@@ -0,0 +1,51 @@
+import { Edge, GraphNode, QueryResult } from './graph.types';
+import { has } from 'dot-prop';
+
+export interface Query {
+  to?: any;
+  from?: any;
+  weights?: {
+    eros?: any;
+    eris?: any;
+    philos?: any;
+    agape?: any;
+  };
+}
+
+const traverseKeys = (o: object, acc: Array<string> = [], parent = '') =>
+  Object.keys(o).reduce((a, k) => {
+    if (typeof o[k] === 'object' && !Array.isArray(o[k])) {
+      return traverseKeys(o[k], a, parent ? `${parent}.${k}` : k);
+    }
+
+    a.push(parent ? `${parent}.${k}` : k);
+    return acc;
+  }, acc);
```

Comment:
could you add a code comment clarifying what this is doing exactly? it's a bit hard to read since reduce isn't the most common along with 1 letter variable names

Reply:
Can do